app/editor/src/store/hooks/editor/useContent.ts

```diff
@@ -30,74 +30,79 @@ export const useContent = (props?: IContentProps): [IContentState, IContentContr
   const dispatch = useAjaxWrapper();
   const api = useApiContents();
 
-  const controller = React.useRef({
-    getContent: async (id: number) => {
-      return (await dispatch('get-content', () => api.getContent(id), 'content')).data;
-    },
-    findContent: async (filter: IContentFilter) => {
-      const response = await dispatch('find-contents', () => api.findContent(filter));
-      actions.storeContent(response.data);
-      return response.data;
-    },
-    addContent: async (content: IContentModel) => {
-      const response = await dispatch('add-content', () => api.addContent(content), 'content');
-      if (state.content) {
-        actions.storeContent({ ...state.content, items: [...state.content.items, content] });
-      }
-      return response.data;
-    },
-    updateContent: async (content: IContentModel) => {
-      const response = await dispatch(
-        'update-content',
-        () => api.updateContent(content),
-        'content',
-      );
-      if (state.content) {
-        actions.storeContent({
-          ...state.content,
-          items: state.content.items.map((i: IContentModel) => {
-            if (i.id === content.id) return content;
-            return i;
-          }),
-        });
-      }
-      return response.data;
-    },
-    deleteContent: async (content: IContentModel) => {
-      const response = await dispatch(
-        'delete-content',
-        () => api.deleteContent(content),
-        'content',
-      );
-      if (state.content) {
-        actions.storeContent({
-          ...state.content,
-          items: state.content.items.filter((i: IContentModel) => i.id !== content.id),
-        });
-      }
-      return response.data;
-    },
-    publishContent: async (content: IContentModel) => {
-      return (await dispatch('publish-content', () => api.publishContent(content), 'content')).data;
-    },
-    unpublishContent: async (content: IContentModel) => {
-      return (await dispatch('unpublish-content', () => api.unpublishContent(content), 'content'))
-        .data;
-    },
-    upload: async (content: IContentModel, file: File) => {
-      return (await dispatch('upload-content', () => api.upload(content, file), 'content')).data;
-    },
-    download: async (id: number, fileName: string) => {
-      return (await dispatch('download-content', () => api.download(id, fileName), 'content')).data;
-    },
-    attach: async (id: number, path: string) => {
-      return (
-        await dispatch<IContentModel>('attach-content', () => api.attach(id, path), 'content')
-      ).data;
-    },
-    storeFilter: actions.storeFilter,
-    storeFilterAdvanced: actions.storeFilterAdvanced,
-  }).current;
+  const controller = React.useMemo(
```

Comment:
`useRef` as causing the issue.  I think I originally attempted to eliminate `useMemo` due to infinite state changes at some point in the past.

Reply:
Ahhh I was close to this at one point lol. Glad you got it figured out. Thanks! :)